src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -987,18 +987,34 @@ private static string EnsurePackageExtension(string packagePath, bool isSnupkg)
         }
 
         /// <summary>
-        /// This method checks of a given list of package paths exist on disk.
+        /// If there isn't at least one Path specified, throw that no file paths were resolved for this Package.
         /// </summary>
-        /// <param name="packagePath">A package path to be used while creating logs.</param>
-        /// <param name="matchingPackagePaths">A list of matching package paths that need to be checked.</param>
+        /// <param name="packagePath">The package path the user originally provided for use while creating logs.</param>
+        /// <param name="matchingPackagePaths">A list of matching package paths that were previously resolved.</param>
         public static void EnsurePackageFileExists(string packagePath, IEnumerable<string> matchingPackagePaths)
         {
-            if (!matchingPackagePaths.Any())
+            if (!PackagePathResolved(matchingPackagePaths))
             {
+                ThrowUnableToFindFile(packagePath);
+            }
+        }
+
+        public static void ThrowUnableToFindFile(string packagePath)
+        {
                 throw new ArgumentException(string.Format(CultureInfo.CurrentCulture,
                     Strings.UnableToFindFile,
                     packagePath));
-            }
+        }
+
+        public static bool PackagePathResolved(string packagePath, bool isSnupkgPush)
```

Comment:
These 2 overrides should start with "Is" to match convention in this class, and to follow these guidelines:

✓ DO name Boolean properties with an affirmative phrase (CanSeek instead of CantSeek). Optionally, you can also prefix Boolean properties with "Is," "Can," or "Has," but only where it adds value.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members

Reply:
I'm fine changing this, but what you quoted is boolean properties. I introduced a boolean method, which in my opinion adheres to "✓ DO give methods names that are verbs or verb phrases." Their example of a verb is `CompareTo`. I could restructure it if we think that `PackagePathResolved` should be `ResolvesToPackagePath` ?

Reply:
Perhaps change the rest of the name too.  Ressolve in ResolvePackageFromPath returns IEnum of string.  In this case, I'm not sure what you mean by Resolved

Reply:
ResolvePackageFromPath is an existing method. I assume what Rohit meant there was that you want to determine if a path resolves to an actual filename.

What I mean by Resolved is: "did it resolve?".

For nits regarding naming, I'd like to be given suggestions. Otherwise we could enter a naming discussion ad infinitum.

Reply:
don't have a great idea for a name...but it is a strange method...  HasAnyStrings(...) would be better...but then it is strange too.

Reply:
one more thing... only one of these overrides is used. delete the one that is not.